src/domain/payments/onchain-payment-flow-builder.ts

```diff
@@ -308,6 +308,25 @@ const OPFBWithAmount = <S extends WalletCurrency, R extends WalletCurrency>(
 const OPFBWithConversion = <S extends WalletCurrency, R extends WalletCurrency>(
   statePromise: Promise<OPFBWithConversionState<S, R> | DealerPriceServiceError>,
 ): OPFBWithConversion<S, R> | OPFBWithError => {
+  const validateBtcPaymentAmountForDust = ({
+    state,
+    btcPaymentAmount,
+  }: {
+    state: OPFBWithConversionState<S, R>
+    btcPaymentAmount: BtcPaymentAmount
+  }): BtcPaymentAmount | DealerPriceServiceError | ValidationError => {
```

Comment:
how can this return `DealerPriceServiceError`?

Reply:
Oversight from the last refactor where I had the full `statePromise` returning in that function. Fixed now.